lib/active_record_host_pool/connection_adapter_mixin.rb

```diff
@@ -64,8 +64,15 @@ def disconnect_with_host_pooling!
 
     private
 
+    attr_accessor :_no_switch
+
     def _switch_connection
-      if _host_pool_current_database && ((_host_pool_current_database != @_cached_current_database) || @connection.object_id != @_cached_connection_object_id)
+      cached = defined?(@_cached_current_database) && @_cached_current_database
```

Comment:
I’d prefer if it was possible to declare `@ _cached_current_database` on initialization.